spec/lib/amplitude_api_spec.rb

```diff
@@ -17,9 +17,11 @@
             event: JSON.generate([event.to_hash])
           }
 
-          expect(Typhoeus).to receive(:post).with(AmplitudeAPI::TRACK_URI_STRING, body: body)
+          request_stub = stub_request(:post, AmplitudeAPI::TRACK_URI_STRING).with(body: body)
```

Comment:
Note that this is not so DRY and not too idiomatic rspec. I didn't want to convert the existing setup steps in minitest style to rspec `let` and `let!`s. Not sure if this was an intentional style choice or not, I can change them if you wish.